hazelcast/src/main/java/com/hazelcast/client/impl/clientside/FailoverClientConfigSupport.java

```diff
@@ -105,11 +105,11 @@ private static ClientFailoverConfig locateAndCreateClientFailoverConfig() {
         YamlClientFailoverConfigLocator yamlConfigLocator = new YamlClientFailoverConfigLocator();
 
         if (xmlConfigLocator.locateFromSystemProperty()) {
-            // 1. Try loading config if provided in system property and it is an XML file
+            // 1. Try loading XML config from the configuration provided in system property
             config = new XmlClientFailoverConfigBuilder(xmlConfigLocator).build();
 
-        } else if (yamlConfigLocator.locateFromSystemPropertyOrFailOnUnacceptedSuffix()) {
```

Comment:
It can only be an accepted YAML suffix here. See the `validateSuffixInSystemProperty` a few lines above. It was validated for XML and YAML suffixes, and it is not XML.